mage/query-modules/cpp/collections.md

```diff
@@ -0,0 +1,486 @@
+
+#### Usage:
+
+```cypher
+CALL collections.min([1,2,3]) YIELD min RETURN min;
+```
+
+```plaintext
++---------------------------------------------------------+
+| min                                                     |
++---------------------------------------------------------+
+| 1                                                       |
++---------------------------------------------------------+
+```
+
+### `unionAll(list1, list2)`
+
+Returns the union of two input lists, including duplicates
+
+#### Input:
+
+- `list1: List[Any]` ➡ the first list to be used in the `unionAll` operation
+- `list2: List[Any]` ➡ the second list to be used in the `unionAll` operation
+
+
+#### Output:
+
+- `return_list: List[Any]` ➡ the union of two lists, including duplicates
+
+#### Usage:
+
+```cypher
+CALL collections.unionAll([1,1,2,3],[3,"a","b","c"]) YIELD return_list RETURN return_list;
+```
+
+```plaintext
++---------------------------------------------------------+
+| return_list                                             |
++---------------------------------------------------------+
+| [1,1,2,3,3,"a","b","c"]                                     |
++---------------------------------------------------------+
+```
+
+### `to_set(list)`
+
+Converts the input list to a set
+#### Input:
+
+- `list: List[Any]` ➡ the list that is to be converted into a set
+
+#### Output:
+
+- `result: List[Any]` ➡ the set returned as an mgp::List data type
+
+#### Usage:
+
+```cypher
+CALL collections.to_set([1,2,1,2,3]) YIELD result RETURN result;
+```
+
+```plaintext
++---------------------------------------------------------+
+| result                                                  |
++---------------------------------------------------------+
+| [3,2,1]                                                 |
++---------------------------------------------------------+
+```
+
+### `partition(list, partition_size)`
+
+Partitions the input list into sub-lists of the specified partition_size
+#### Input:
+
+- `list: List[Any]` ➡ the list that is to be partitioned
+- `partition_size: integer` ➡ the size of the sub-lists
+
+#### Output:
+
+- `result: List[Any]` ➡ the partitioned sub-lists
+
+#### Usage:
+
+```cypher
+CALL collections.partition([1,2,3,4,5,6],2) YIELD result RETURN result;
+```
+
+```plaintext
++---------------------------------------------------------+
+| result                                                  |
++---------------------------------------------------------+
+| [1,2]                                                   |
++---------------------------------------------------------+
+| [3,4]                                                   |
++---------------------------------------------------------+
+| [5,6]                                                   |
++---------------------------------------------------------+
+```
+
+### `sum_longs(numbers)`
+
+Calculates the sum of list elements casted to integers.
```

Comment:
casted?

Reply:
Casting means converting one data type into the other.

Reply:
you live, you learn

What if you never cast them and they were numbers all along?

Reply:
Im looking at the code right now, they will be casted to integer no matter what, if they were numbers before, everything is good, if they were some other type, etc. string ot list, an exception is thrown, I will add that to description